packages/utils/tests/get-snark-artifacts.node.test.ts

```diff
@@ -0,0 +1,127 @@
+import { createWriteStream, existsSync } from "node:fs"
+import { mkdir } from "node:fs/promises"
+import { getEddsaSnarkArtifacts, getPoseidonSnarkArtifacts } from "../src/get-snark-artifacts.node"
+
+jest.mock("node:fs", () => ({
+    ...jest.requireActual("node:fs"),
+    createWriteStream: jest.fn(),
+    existsSync: jest.fn()
+}))
+
+jest.mock("node:fs/promises", () => ({
+    ...jest.requireActual("node:fs/promises"),
+    mkdir: jest.fn()
+}))
+
+global.fetch = jest.fn()
+
+beforeEach(() => {
+    jest.resetAllMocks()
+    ;(mkdir as jest.Mock).mockResolvedValue(undefined)
+})
+
+describe("getPoseidonSnarkArtifacts", () => {
+    it("should handle fetch errors", async () => {
+        ;(existsSync as jest.Mock).mockReturnValue(false)
+        ;(global.fetch as jest.Mock).mockResolvedValueOnce({
+            ok: false,
+            statusText: "TEST"
+        })
+        await expect(getPoseidonSnarkArtifacts(2)).rejects.toThrowErrorMatchingInlineSnapshot(
```

Comment:
needs jest >= 30.  
see https://github.com/privacy-scaling-explorations/zk-kit/pull/240/files#r1556087134

somehow I could not make it work with [`prettierPath`](https://jestjs.io/docs/configuration#prettierpath-string)